climada/hazard/trop_cyclone.py

```diff
@@ -1090,31 +1177,166 @@ def compute_angular_windspeeds(si_track, d_centr, mask_centr_close, model, cyclo
     -------
     ndarray of shape (npositions, ncentroids)
     """
-    if model == MODEL_VANG['H1980']:
-        _B_holland_1980(si_track)
-    elif model in [MODEL_VANG['H08'], MODEL_VANG['H10']]:
-        _bs_holland_2008(si_track)
-
-    if model in [MODEL_VANG['H1980'], MODEL_VANG['H08']]:
-        result = _stat_holland_1980(
-            si_track, d_centr, mask_centr_close, cyclostrophic=cyclostrophic,
-        )
-        if model == MODEL_VANG['H1980']:
-            result *= GRADIENT_LEVEL_TO_SURFACE_WINDS
-    elif model == MODEL_VANG['H10']:
-        # this model is always cyclostrophic
-        _v_max_s_holland_2008(si_track)
-        hol_x = _x_holland_2010(si_track, d_centr, mask_centr_close)
-        result = _stat_holland_2010(si_track, d_centr, mask_centr_close, hol_x)
-    elif model == MODEL_VANG['ER11']:
-        result = _stat_er_2011(si_track, d_centr, mask_centr_close, cyclostrophic=cyclostrophic)
-    else:
-        raise NotImplementedError
-
+    model_kwargs = {} if model_kwargs is None else model_kwargs
+    compute_funs = {
+        MODEL_VANG['H1980']: _compute_angular_windspeeds_h1980,
+        MODEL_VANG['H08']: _compute_angular_windspeeds_h08,
+        MODEL_VANG['H10']: _compute_angular_windspeeds_h10,
+        MODEL_VANG['ER11']: _stat_er_2011,
+    }
+    if model not in compute_funs:
+        raise NotImplementedError(f"The specified wind model is not supported: {model}")
+    result = compute_funs[model](
+        si_track,
+        d_centr,
+        mask_centr_close,
+        cyclostrophic=cyclostrophic,
+        **model_kwargs,
+    )
     result[0, :] *= 0
```

Comment:
I assume what happens is that for the frist track position this is set to zero. If this is true, I do not know why this is done here as it will be done again anyway in Line ~1051: `windfields[0, :, :] = 0`.
I personally would not remove this line here, just to repeat it in 4 different places (the `compute_funs`).  I would leave it as it is.